scripts/zkai_one_block_executable_package_accounting_gate.py

```
@@ -0,0 +1,660 @@
+        source_descriptor(COMPRESSION, "statement_chain_compression_json", compression_raw, compression),
+        source_descriptor(SNARK_RECEIPT, "snark_statement_receipt_json", snark_raw, snark),
+    ]
+    return scorecard, compression, snark, sources
+
+
+def package_summary(compression: dict[str, Any], snark: dict[str, Any]) -> dict[str, Any]:
+    compression_summary = _dict(compression.get("summary"), "compression.summary")
+    snark_receipt = _dict(snark.get("receipt_metrics"), "snark.receipt_metrics")
+    source_bytes = _int(compression_summary.get("source_chain_artifact_bytes"), "source bytes")
+    if source_bytes <= 0:
+        raise OneBlockPackageAccountingError("source bytes must be positive")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)